core/src/main/java/hudson/DNSMultiCast.java

```diff
@@ -1,205 +1,20 @@
-         * Works like {@link #close()} but terminate uncleanly, but rapidly and without blocking.
-         */
-        public void abort() throws IOException {
-            if (this.isClosing()) {
-                return;
-            }
-
-           if (logger.isLoggable(Level.FINER)) {
-                logger.finer("Aborting JmDNS: " + this);
-            }
-            // Stop JmDNS
-            // This protects against recursive calls
-            if (this.closeState()) {
-                // We got the tie break now clean up
-
-                // Stop the timer
-                logger.finer("Canceling the timer");
-                this.cancelTimer();
-
-                // Cancel all services
-                // KK: this is a blocking call that doesn't fit 'abort'
-                // this.unregisterAllServices();
-                executePrivateParentMethod("disposeServiceCollectors");
-
-// KK: another blocking call
-//                if (logger.isLoggable(Level.FINER)) {
-//                    logger.finer("Wait for JmDNS cancel: " + this);
-//                }
-//                this.waitForCanceled(DNSConstants.CLOSE_TIMEOUT);
-
-                // Stop the canceler timer
-                logger.finer("Canceling the state timer");
-                this.cancelStateTimer();
-
-                // Stop the executor
-                shutdown();
-
-                // close socket
-                executePrivateParentMethod("closeMulticastSocket");
-
-                // remove the shutdown hook
-                if (_shutdown != null) {
-                    Runtime.getRuntime().removeShutdownHook(_shutdown);
-                }
-
-                if (logger.isLoggable(Level.FINER)) {
-                    logger.finer("JmDNS closed.");
-                }
-            }
-            advanceState(null);
-        }
-
-        private void shutdown() throws IOException {
-            try {
-                Field executor = this.parent.getDeclaredField("_executor");
-                executor.setAccessible(true);
-                ExecutorService _executor = (ExecutorService) executor.get(this);
-                _executor.shutdown();
-            } catch (NoSuchFieldException | IllegalAccessException e) {
-                logger.log(Level.SEVERE, "Error trying to abort JmDNS", e);
-                throw new IOException(e);
-            }
-        }
-
-        private void executePrivateParentMethod(String method) throws IOException {
-            try {
-                Method m = this.parent.getDeclaredMethod(method);
-                m.setAccessible(true);
-                m.invoke(this);
-            } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
-                logger.log(Level.SEVERE, "Error trying to abort JmDNS", e);
-                throw new IOException(e);
-            }
-        }
+    public static boolean disabled = SystemProperties.getBoolean(DNSMultiCast.class.getName()+".disabled", true);
```

Comment:
Might be worth marking this field (or the class in general) as deprecated?

Reply:
Thanks. I had marked it as deprecated at one point but ended up losing that in the final submission.